Mailer/app/api/v1/Handlers/Users/UserPreferencesHandler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Remp\MailerModule\Api\v1\Handlers\Users;
+
+use Remp\MailerModule\Api\Validation\InputValidatorTrait;
+use Remp\MailerModule\Repository\UserSubscriptionsRepository;
+use Remp\MailerModule\Repository\UserSubscriptionVariantsRepository;
+use Tomaj\NetteApi\Handlers\BaseHandler;
+use Tomaj\NetteApi\Response\JsonApiResponse;
+
+class UserPreferencesHandler extends BaseHandler
+{
+    private $userSubscriptionsRepository;
+
+    private $userSubscriptionVariantsRepository;
+
+    use InputValidatorTrait;
+
+    public function __construct(
+        UserSubscriptionsRepository $userSubscriptionsRepository,
+        UserSubscriptionVariantsRepository $userSubscriptionVariantsRepository
+    ) {
+        parent::__construct();
+        $this->userSubscriptionsRepository = $userSubscriptionsRepository;
+        $this->userSubscriptionVariantsRepository = $userSubscriptionVariantsRepository;
+    }
+
+    public function handle($payload)
+    {
+
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'user-preferences.schema.json');
+
+        if ($this->hasErrorResponse()){
+            return $this->getErrorResponse();
+        }
+
+//        $rows = $this->userSubscriptionsRepository->findByEmail($payload['email']); TODO use later
```

Comment:
What with this?

Reply:
In future identification of the user should be based only on email, not on a combination of email and id.

Reply:
Oh, in that case please change the TODO to something that we'll be able to understand when we get to it.. "TODO: use this once mailer uses email as a primary identifier" or something like this.